test/e2e/e2e-test/src/test/java/org/apache/skywalking/e2e/LuaE2E.java

```diff
@@ -92,14 +95,21 @@
     public void setUp() throws Exception {
         queryClient(swWebappHostPort);
 
-        trafficController(entryProvider, "/nginx/entry/info?backend=" + nginxHostPort.host() + ":" + nginxHostPort.port());
+        trafficController(entryProvider, "/nginx/entry/info");
     }
 
     @AfterAll
     public void tearDown() {
         trafficController.stop();
     }
 
+    @RetryableTest
+    public void correlation() throws Exception {
+        final URL url = new URL("http", entryProvider.host(), entryProvider.port(), "/nginx/entry/info");
+        final ResponseEntity<String> response = restTemplate.postForEntity(url.toURI(), trafficData, String.class);
+        Assert.assertEquals(response.getBody(), "entry_value_nginx_value");
```

Comment:
@mrproliu Do you mean this check?

Reply:
Yes, there will check all the correlation data from `java2` response.

Reply:
So, once the propagation fails, this e2e will fail directly. Got it.